platform/android/java/app/config.gradle

```diff
@@ -1,11 +1,11 @@
 ext.versions = [
-    androidGradlePlugin: '8.2.0',
-    compileSdk         : 34,
-    minSdk             : 19, // Also update 'platform/android/export/export_plugin.cpp#DEFAULT_MIN_SDK_VERSION'
-    targetSdk          : 34, // Also update 'platform/android/export/export_plugin.cpp#DEFAULT_TARGET_SDK_VERSION'
-    buildTools         : '34.0.0',
-    kotlinVersion      : '1.9.20',
-    fragmentVersion    : '1.6.2',
+    androidGradlePlugin: '8.6.1',
+    compileSdk         : 35,
+    minSdk             : 21, // Also update 'platform/android/export/export_plugin.cpp#DEFAULT_MIN_SDK_VERSION'
+    targetSdk          : 35, // Also update 'platform/android/export/export_plugin.cpp#DEFAULT_TARGET_SDK_VERSION'
+    buildTools         : '35.0.0',
+    kotlinVersion      : '2.1.20',
+    fragmentVersion    : '1.8.6',
     nexusPublishVersion: '1.3.0',
     javaVersion        : JavaVersion.VERSION_17,
     ndkVersion         : '23.2.8568313' // Also update 'platform/android/detect.py#get_ndk_version' when this is updated.
```

Comment:
The ndk also needs to be updated in order to add support for device with 16kb page.
See https://github.com/godotengine/godot/pull/105611, https://github.com/godotengine/godot/pull/106328, and https://github.com/godotengine/godot/pull/106358.

Reply:
Okay, I believe I've gotten everything that needs to be done in there, but please check again with a harder look, because, per my notes below, there were some changes I wasn't sure on that appeared to be 4.x-only, and I wanted to keep the changes as minimal as possible.

I kept it in a separate commit so you could see these specific changes more easily. Also, I did test all forms of exports and running directly on my device and it seemed to work in all cases, but I have not gotten permission from my team yet to test it on an exported Android build in our internal testing track on Play. If I can get that permission, I will definitely try that as well.